test/SlowTests/Issues/RavenDB-17451.cs

```diff
@@ -0,0 +1,233 @@
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading.Tasks;
+using FastTests.Server.Replication;
+using Raven.Client.Documents.Subscriptions;
+using Raven.Tests.Core.Utils.Entities;
+using SlowTests.Core.AdminConsole;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_17451 : ReplicationTestBase
+    {
+        public RavenDB_17451(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task CanDisableTcpCompressionViaConfiguration_ReplicationTest(bool disableOnSrc)
+        {
+            var srcServer = GetNewServer(new ServerCreationOptions
+            {
+                RunInMemory = false
+            });
+            var dstServer = GetNewServer(new ServerCreationOptions
+            {
+                RunInMemory = false
+            });
+            using (var srcStore = GetDocumentStore(new Options
+            {
+                Server = srcServer
+            }))
+            using (var dstStore1 = GetDocumentStore(new Options
+            {
+                Server = dstServer
+            }))
+            using (var dstStore2 = GetDocumentStore(new Options
+            {
+                Server = dstServer
+            }))
+            {
+                srcServer.ServerStore.LicenseManager.LicenseStatus.Attributes["tcpDataCompression"] = true;
+                dstServer.ServerStore.LicenseManager.LicenseStatus.Attributes["tcpDataCompression"] = true;
+
+                const string docId = "users/1";
+                using (var session = srcStore.OpenSession())
+                {
+                    session.Store(new User { Name = "ayende" }, docId);
+                    session.SaveChanges();
+                }
+
+                await SetupReplicationAsync(srcStore, dstStore1);
+
+                Assert.True(WaitForDocument<User>(dstStore1, docId, u => u.Name == "ayende"));
+
+                var srcDb = await srcServer.ServerStore.DatabasesLandlord.TryGetOrCreateResourceStore(srcStore.Database);
+
+                var tcpConnections = srcDb.RunningTcpConnections.ToList();
+
+                Assert.Equal(1, tcpConnections.Count);
+
+                var repToDst1TcpInfo = tcpConnections[0];
+
+                // assert compressed tcp connection to dst1
+                Assert.True(repToDst1TcpInfo.Stream is Sparrow.Utils.ReadWriteCompressedStream);
+
+                var serverToDisableOn = disableOnSrc ? srcServer : dstServer;
+                var configuration = serverToDisableOn.Configuration;
+                Assert.False(configuration.Databases.DisableTcpCompression);
+
+                // modify configuration
+                AdminJsConsoleTests.ExecuteScript(serverToDisableOn, database: null, "server.Configuration.Databases.DisableTcpCompression = true;");
```

Comment:
Why do you set it this way? You can modify the configuration option via `GetNewServer()` with the usage of `ServerCreationOptions.CustomSettings`